GOVERNANCE_DOC.md

```diff
@@ -0,0 +1,597 @@
+      // InitialDeposit is negative or null OR sender has insufficient funds
+      
+      throw
+    
+    else
+      sender.AtomBalance -= InitialDeposit
+      
+      proposalID = generate new proposalID
+      proposal = create new Proposal from proposalID
+      
+      proposal.Title = txSubmitProposal.Title
+      proposal.Description = txSubmitProposal.Description
+      proposal.Type = txSubmitProposal.Type
+      proposal.Category = txSubmitProposal.Category
+      proposal.Deposit = txSubmitProposal.InitialDeposit
+      proposal.SubmitBlock = CurrentBlock
+      
+      create depositorsList from proposalID
+      initiate deposit of sender in depositorsList at txSubmitProposal.InitialDeposit
+  
+      if (txSubmitProposal.InitialDeposit < ActiveProcedure.MinDeposit) then  
+        // MinDeposit is not reached
+        
+        proposal.VotingStartBlock = -1
+      
+      else  
+        // MinDeposit is reached
+        
+        proposal.VotingStartBlock = CurrentBlock
+        
+        create  votersList,
+                initVotingPowerList,
+                minusesList,
+                initProcedureNumber,
+                initTotalVotingPower  from proposalID
+                                    
+        snapshot(ActiveProcedure.ProcedureNumber) // Save current procedure number in initProcedureNumber
+        snapshot(TotalVotingPower)  // Save total voting power in initTotalVotingPower
+        snapshot(ValidatorVotingPower)  // Save validators' voting power in initVotingPowerList
+        
+        ProposalProcessingQueueEnd++
+        ProposalProcessingQueue[ProposalProcessingQueueEnd] = proposalID
+  
+      return proposalID
+```
+
+And the pseudocode for the `ProposalProcessingQueue`:
+
+```
+
+  in BeginBlock do 
+    
+    checkProposal()  
+    
+    
+    
+  func checkProposal()  
+    if (ProposalProcessingQueueBeginning == ProposalProcessingQueueEnd)
+      return
+
+    else
+      retrieve proposalID from ProposalProcessingQueue[ProposalProcessingQueueBeginning]
+      retrieve proposal from proposalID
+      retrieve initProcedureNumber from proposalID
+      retrieve initProcedure from initProcedureNumber
+
+      if (CurrentBlock == proposal.VotingStartBlock + initProcedure.VotingPeriod)
+        retrieve initVotingPowerList from proposalID
+        retrieve votersList from proposalID
+        retrieve validators from initVotingPowerList
+
+        for each validator in validators
+          if validator is not in votersList
+            slash validator by ActiveProcedure.GovernancePenalty
+
+        ProposalProcessingQueueBeginning++  // ProposalProcessingQueue will have a new element
+        checkProposal()
+
+      else
+        return
+          
+```
+
+Once a proposal is submitted, if `Proposal.Deposit < ActiveProcedure.MinDeposit`, Atom holders can send `TxDeposit` transactions to increase the proposal's deposit.
+
+```Go
+
```

Comment:
can delete these extra spaces in the spec within the code seperators hehe

Reply:
ok